packages/web-forms/src/components/common/map/useMapBlock.ts

```diff
@@ -0,0 +1,282 @@
+			title: 'Cannot access location',
+			message:
+				'Grant location permission in the browser settings and make sure location is turned on.',
+		};
+
+		if (!navigator.geolocation) {
+			currentState.value = STATES.ERROR;
+			errorMessage.value = friendlyError;
+		}
+
+		navigator.geolocation.getCurrentPosition(
+			(position) => {
+				const coords = fromLonLat([position.coords.longitude, position.coords.latitude]);
+				mapInstance
+					?.getView()
+					.animate({ center: coords, zoom: MAX_ZOOM, duration: ANIMATION_TIME });
+				currentState.value = STATES.READY;
+			},
+			() => {
+				currentState.value = STATES.ERROR;
+				errorMessage.value = friendlyError;
+			},
+			{ enableHighAccuracy: true, timeout: GEOLOCATION_TIMEOUT_MS }
+		);
+	};
+
+	const centerFeatureLocation = (feature: Feature<GeometryType>): void => {
+		const geometry = feature.getGeometry();
+		const size = mapInstance?.getSize();
+		const view = mapInstance?.getView();
+
+		if (!geometry || !size?.length || !view) {
+			return;
+		}
+
+		const width = size[0];
+		const height = size[1];
+		let pixelOffsetY = 0;
+		let pixelOffsetX = -50;
+		if (width < SMALL_DEVICE_WIDTH) {
+			pixelOffsetY = 130;
+			pixelOffsetX = 0;
+		}
+
+		view.centerOn(geometry.getExtent(), size, [
+			width / 2 - pixelOffsetX,
+			height / 2 - pixelOffsetY,
+		]);
+	};
+
+	const loadGeometries = (geoJSON: FeatureCollection): void => {
+		if (!mapInstance) {
+			return;
+		}
+
+		currentState.value = STATES.LOADING;
+		unselectFeature();
+		discardSavedFeature();
+		featuresSource.clear();
+
+		if (!geoJSON.features.length) {
+			mapInstance?.getView().animate({
+				center: DEFAULT_VIEW_CENTER,
+				zoom: MIN_ZOOM,
+				duration: ANIMATION_TIME,
+			});
+			currentState.value = STATES.READY;
+			return;
+		}
+
+		const features = new GeoJSON().readFeatures(geoJSON, {
+			dataProjection: DEFAULT_GEOJSON_PROJECTION,
+			featureProjection: mapInstance.getView().getProjection(),
+		});
+
+		features.forEach((feature) => {
+			if (!feature.get(FEATURE_ID_PROPERTY)) {
+				feature.set(FEATURE_ID_PROPERTY, crypto.randomUUID());
+			}
+		});
+		featuresSource.addFeatures(features);
+		currentState.value = STATES.READY;
+
+		fitToAllFeatures();
+	};
+
+	const selectFeatureByPosition = (position: Pixel): void => {
+		const hitFeatures = mapInstance?.getFeaturesAtPixel(position, {
+			hitTolerance: MAP_HIT_TOLERANCE,
+			layerFilter: (layer) => layer instanceof WebGLVectorLayer,
+		});
+
+		const featureToSelect = hitFeatures?.length
+			? (hitFeatures[0] as Feature<GeometryType>)
+			: undefined;
+
+		selectFeature(featureToSelect);
+	};
+
+	const selectFeature = (feature: Feature<GeometryType> | undefined): void => {
+		selectedFeature.value = feature;
+
+		featuresVectorLayer.updateStyleVariables({
+			[SELECTED_ID_PROPERTY]: (selectedFeature.value?.get(FEATURE_ID_PROPERTY) as string) ?? '',
+		});
+
+		if (selectedFeature.value != null) {
+			centerFeatureLocation(selectedFeature.value);
+		}
```

Comment:
I _think_ this would be cleaner if it was updated reactively based on the `selectedFeature.value`, and ditto for the next 3 functions.

Then the `selectFeature` function would just be one line and could be inlined where it's used, right?